### PR TITLE
[Wallet] Split doVerification saga and UI changes to verification

### PR DESCRIPTION
**Pull request by i1skn**
### Description

This allows to fetch current verification state first, then give user options on how to continue and only then verification starts.

This is how current verification flows are affected:
* Onboarding via **invite** - **no changes**
* Onboarding via **backup phrase** - we check the current verification state in the end of restoring account process, then we show `VerificationEducationScreen` as previously (so it takes a little longer for user to see verification screen, but verification screen **does not show a loader**).
* User goes **to Verification screen from inside the app** - we show a loader and meanwhile we check the current verification state, then we show `VerificationEducationScreen` as previously (so it takes a little longer for user to see verification screen and verification screen **does show a loader**).

Depend on the verification state, there are following possible states:
* User has 0 completed and 0 not expired attestations - we show button `Start`
* User do not have sufficient balance to perform verification - appropriate message and button `Skip it now` would appear
*  User has `>0` completed and `0` not expired attestations - we show button `Resume`
*  User has the same amount of attestations left to complete and not expired ones, which both `>0` - we show two buttons -  `Resume` and `I already received three codes`. The difference is that we **skip the revealing step** for attestations in case of `I already received three codes` button is pressed.

### Other changes

Change right CTA on verification notification from `Remind` to `Dismiss`

### Tested

iOS simulator (have tried to cover all the flow described above)

### Related issues

- Part of https://github.com/celo-org/celo-monorepo/issues/4563
- Part of https://github.com/celo-org/celo-labs/issues/566

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/mobile/src/identity/actions.ts [hunk 1]
```diff
@@ -198,9 +211,12 @@ export type ActionTypes =
   | EndFetchingAddressesAction
   | FetchDataEncryptionKeyAction
   | UpdateAddressDekMapAction
+  | FetchVerificationState
+  | UpdateVerificationState
 
-export const startVerification = (): StartVerificationAction => ({
+export const startVerification = (withoutRevelaling: boolean = false): StartVerificationAction => ({
```
Comment (gnardini) [inline packages/mobile/src/identity/actions.ts]: There's a typo in the variable name `withoutRevelaling`

**Review by gnardini — APPROVED** [review]
I'm not very familiar with the attestations code, but everything looks good to me, just left a small nit :)

**Review by annakaz — APPROVED** [review]
LGTM 🚀

File: packages/mobile/src/config.ts [hunk 2]
```diff
@@ -55,6 +55,7 @@ export const SMS_RETRIEVER_APP_SIGNATURE = Config.SMS_RETRIEVER_APP_SIGNATURE
 // ODIS minimum dollar balance for pepper quota retrieval
 // TODO change this to new ODIS minimum dollar balance once deployed
 export const ODIS_MINIMUM_DOLLAR_BALANCE = 0.1
+export const VERIFICATION_STATE_EXPIRY_SECONDS = 30
```
Comment (aslawson) [inline packages/mobile/src/config.ts:62]: does this relate to the expiry the attestation request?  i'm guess no considering this is so short

Comment (i1skn) [reply packages/mobile/src/config.ts:62]: No, when user goes to initial verification screen - we fetch current verification state. Then user decides on what to do and go to verification itself. If user were "deciding" for more than `VERIFICATION_STATE_EXPIRY_SECONDS` - we would refetch verification status before going to actual verification. I will add it as a comment in the code

**Comment by Lss-Ankit** [comment]
Hi @i1skn , @jeanregisser i have verified task using Android internal build v1.1.0 (1004294309)& iOS test flight build v1.1.0(21) 

**Test scenarios:**
 
- **On-boarding via backup phrase:** Verify Loading indicator is not shown after restoring the backup key. 
**[Attachment:](https://drive.google.com/file/d/1OZ3ateRamGuTGjdLuWOt9A1iIskJejjj/view?usp=sharing)** 
**can you please confirm is it expected?** 

- **User goes to Verification screen from inside the app:** Verify loading indicator is shown after tapping on "Confirm Phone Number" option from Setting screen. 
**[Attachment:](https://drive.google.com/file/d/1MHOSKDD-hMZs1hMTeB0a1ODO9fE7ro_o/view?usp=sharing)**
- Verify "Confirm phone number" screen is shown with "Your current balance is not high enough..." description when user sign in with 0 balance 
- Verify Skip For now and Skip button is shown on the top right corner. 
- Verify Phone Number Confirmation pop up is open when user taps on "Do i Need confirm?" button from Confirm page screen if user has 0 balance.  
- Verify user "Start" button is shown when user has not proceeded for the attestations. 
- Verify Resume and I already received three codes button are shown when user skip the Phone number verification flow after getting three attestations.
- Verify User is redirect to Confirm screen with three fields when user taps on I already received three codes button 
- Verify User is redirect to Confirm screen with three fields when user taps on Resume button. 


 **Need clarifications:** 
1. Skip for now and skip button at top right corner is shown on Confirm phone number screen when user sign in with 0 balance account. 
Can you please confirm is it expected that Skip for now and Skip both button is shown on same page and functionality is also same to skip the phone number verification flow? 
**[Attachment:](https://drive.google.com/file/d/1FsMvvqL3CC7zmG7eldZIsZSCUD7rJFBa/view?usp=sharing)** 
2. Resume and I already received three codes buttons are functioning same when user has balance and taps on both button vise versa. 
**[Attachment:](https://drive.google.com/file/d/1pkJVgztKkIzN4w4-CspSFMgw8apb4aBK/view?usp=sharing)** 

Thanks!

**Pull request merged**